File: tests/cppunit-1.8.0/src/fdcppunit/FiboTest/FiboTestSuite.cpp
#include <cppunit/ui/text/TestRunner.h>
#include <cppunit/TestResultCollector.h>
#include <cppunit/TextTestResult.h>
#include "../FrontDeskOutputter.h"
#include "../FrontDeskTestCaller.h"
#include "../FrontDeskTestSuite.h"
#include <iostream>

#include "FiboTestSuite.h"

using namespace CppUnit::TextUi;
using namespace FDCppUnit;

string FiboTestSuite::m_msg="";

FrontDeskTestSuite* FiboTestSuite::suite() {

	FrontDeskTestSuite* tsuite = new FrontDeskTestSuite("FiboTestSuite");

	tsuite->addTest(new FrontDeskTestCaller<FiboTestSuite>("testF10", &FiboTestSuite::testF10, 4.0, 20.0, 5));
	tsuite->addTest(new FrontDeskTestCaller<FiboTestSuite>("testF8", &FiboTestSuite::testF8, 4.0, 20.0, 5));
	tsuite->addTest(new FrontDeskTestCaller<FiboTestSuite>("testF12", &FiboTestSuite::testF12, 4.0, 20.0, 5));
	tsuite->addTest(new FrontDeskTestCaller<FiboTestSuite>("testF1", &FiboTestSuite::testF1, 4.0, 20.0, 5));
	tsuite->addTest(new FrontDeskTestCaller<FiboTestSuite>("testF0", &FiboTestSuite::testF0, 4.0, 20.0, 5));

	return tsuite;
}

string* FiboTestSuite::getMessage() {
	return &m_msg;
}

void FiboTestSuite::setUp() { }

void FiboTestSuite::tearDown() { }

void FiboTestSuite::testF10() {
	m_msg += "F(10) testing...\n";
	CPPUNIT_ASSERT_MESSAGE("F<10> test failed", m_fibo.fibo(10) == 89);
}

void FiboTestSuite::testF8() {
	m_msg += "F(8) testing...\n";
	CPPUNIT_ASSERT_MESSAGE("F<<8> test failed", m_fibo.fibo(8) == 34);
}

void FiboTestSuite::testF12() {
	m_msg += "F(12) testing...\n";
	CPPUNIT_ASSERT_MESSAGE("F<12> test failed", m_fibo.fibo(12) == 233);
}

void FiboTestSuite::testF1() {
	m_msg += "F(1) testing...\n";
	CPPUNIT_ASSERT_MESSAGE("F<1> test failed", m_fibo.fibo(1) == 1);
}

void FiboTestSuite::testF0() {
	m_msg += "F(0) testing...\n";
	CPPUNIT_ASSERT_MESSAGE("F<0> test failed", m_fibo.fibo(0) == 1);
}

int main() {

	TestRunner runner;
	FrontDeskTestSuite* suite = FiboTestSuite::suite();
	runner.addTest(suite);

	runner.setOutputter(new FrontDeskOutputter(&runner.result(), cout, 
								FiboTestSuite::getMessage(), suite->getElapsed()));
	runner.run("", false, true, false);

	return 0;
}

File: tests/fibo/test/FiboTestSuite.java
//Mike Maxim
//Fibonacci test suite

import junit.framework.*;
import junit.frontdesk.*;
import junit.textui.*;

public class FiboTestSuite extends FrontDeskTestCase {

    public FiboTestSuite() { }

    public FiboTestSuite(String name, double fpoints, double epoints, int timelimit) {
	super(name, fpoints, epoints, timelimit);
    }

    protected Fibo fibo = new Fibo();

    public static Test suite() {
	
	TestSuite suite= new TestSuite();
	suite.addTest(new FiboTestSuite("F(10)", 4.0, 20.0, 5) {
			      public void runTest() { testF10(); } } );
	suite.addTest(new FiboTestSuite("F(8)", 4.0, 20.0, 5) {
			      public void runTest() { testF8(); } } );
	suite.addTest(new FiboTestSuite("F(12)", 4.0, 20.0, 5) {
			      public void runTest() { testF12(); } } );
	suite.addTest(new FiboTestSuite("F(1)", 4.0, 20.0, 5) {
			      public void runTest() { testF1(); } } );
	suite.addTest(new FiboTestSuite("F(0)", 4.0, 20.0, 5) {
			      public void runTest() { testF0(); } } );
	
	return suite;
    }

    public void testF10() {
	assertEquals("F(10)", fibo.fibo(10), 89);
	setCompetitionScore(10);
    }

    public void testF8() {
	assertEquals("F(8)", fibo.fibo(8), 34);
	setCompetitionScore(12);
    }

    public void testF12() {
	assertEquals("F(12)", fibo.fibo(12), 233);
    }

    public void testF1() {
	assertEquals("F(1)", fibo.fibo(1), 1);
    }

    public void testF0() {
	assertEquals("F(0)", fibo.fibo(0), 1);
    }
}
